src/requestManager.cpp

```diff
@@ -515,85 +515,90 @@ bool CRequestManager::RequestBlock(CNode *pfrom, CInv obj)
         }
     }
 
+
     // Ask for XTHIN's if Graphene is not enabled, or, ask for XTHIN's if graphene is enabled
     // but the grapheneblock timer has lapsed.
-    if ((IsThinBlocksEnabled() && IsChainNearlySyncd() && !IsGrapheneBlockEnabled()) ||
-        (IsThinBlocksEnabled() && IsChainNearlySyncd() && IsGrapheneBlockEnabled() &&
-            !graphenedata.CheckGrapheneBlockTimer(obj.hash)))
+    if (IsChainNearlySyncd() && IsThinBlocksEnabled() && HaveThinblockNodes())
     {
-        if (HaveThinblockNodes() && thindata.CheckThinblockTimer(obj.hash))
+        if (!IsGrapheneBlockEnabled() || (IsGrapheneBlockEnabled() && !graphenedata.CheckGrapheneBlockTimer(obj.hash)))
```

Comment:
```cpp
if (!IsGrapheneBlockEnabled() || !graphenedata.CheckGrapheneBlockTimer(obj.hash))
```
redundant logic here, a single `IsGrapheneBlockEnabled()` check is sufficient.

Reply:
@gubatron 

think you are right. good catch. :smile:

Reply:
ok removed the redundant IsGrapheneBlockEnabled()